main.js

```diff
@@ -0,0 +1,161 @@
+//     { productName: "eggs", productPrice: 4, productQuantity: 10, },
+//     { productName: "cheese", productPrice: 110, productQuantity: 4, },
+//     { productName: "tomato", productPrice: 10, productQuantity: 6, },
+//     { productName: "onion", productPrice: 5, productQuantity: 1, },
+// ];
+
+// let totalSum = 0;
+// const totalPrice = shopList.forEach((element) => totalSum += element.productPrice * element.productQuantity);
+// console.log(`Загальна вартість корзини становить ${totalSum} UAH`);
+
+// const justProductQuntity = shopList.map(item => item.productQuantity);
+// const minQuantity = Math.min.apply(null, justProductQuntity);
+
+// function totalPoductQuantity(justProductQuntity) {
+//     let sum = 0;
+//     for (let i = 0; i < justProductQuntity.length; i++) {
+//         sum += justProductQuntity[i];
+//     }
+//     return sum;
+// };
+// console.log(`Загальна кількість продуктів ${totalPoductQuantity(justProductQuntity)} шт.`);
+
+// let mostExpenciveProduct = shopList.reduce((min, max) => min.productPrice > max.productPrice ? min : max);
+// console.log(`Найбіль дорогим продуктом в списку є ${mostExpenciveProduct.productName}!`);
+
+// const addItemInShopList = () => {
+//     const createNewProduct = (productName, productPrice, productQuantity) => {
+//         return {
+//             productName: productName,
+//             productPrice: productPrice,
+//             productQuantity: productQuantity,
+//         }
+
+//     }
+//     const newProduct = createNewProduct(
+//         productName = prompt('Введіть назву продукту:'),
+//         productPrice = +prompt('Введіть ціну продукту:'),
+//         productQuantity = +prompt('Введіть кількість продукту'),
+//     );
+
+//     if (productName === undefined || productName === null || typeof productName != "string") {
+//         console.log('Введені дані не коректні');
+//     }
+//     if (productPrice === undefined || productPrice === null || isNaN(productPrice) || productName === 0) {
+//         console.log('Введені дані не коректні');
+//     }
+//     if (productQuantity === undefined || productQuantity === null || isNaN(productQuantity) || productName === 0) {
+//         console.log('Введені дані не коректні');
+//     }
+//     shopList.push(newProduct);
+
+// }
+// addItemInShopList();
+
+
+
+
+// function deleteExcessProduct() {
+//     let anyProduct = prompt('Що викинемо?');
+//     excessProduct = shopList.find(item => item.productName === anyProduct);
+
+//     let excessIndex = shopList.indexOf(excessProduct);
+//     if (excessIndex !== -1) {
+//         shopList.splice(excessIndex, 1);
+//     }
+// }
+    
+// deleteExcessProduct();
+// console.log(shopList);
+
+// Tesk #4
+
+// ourArray = [16, -3, 54, -4, 72, -56, 47, -12, 4, -16, 25, -37, 46, 4, -51, 27, -8, 4, -54, 76, -4, 12, 6, -35];
+
+// function positiveElements() {
+//     let sumValue = 0;
+//     ourArray.forEach((element) => sumValue += element > 0 ? element : 0);

```

Comment:
We can firstly identify all positive numbers, and then calculate the sum, to avoid making the operation of filtering by `> 0` conditions twice